src/smolagents/default_tools.py

```
@@ -250,11 +250,11 @@ def forward(self, url: str) -> str:
             return truncate_content(markdown_content, 10000)
 
         except requests.exceptions.Timeout:
-            return "The request timed out. Please try again later or check the URL."
+            raise Exception("The request timed out. Please try again later or check the URL.")
```

@aymeric-roucher 

Once again I think it is way better to raise an exception instead of returning an error like a normal response. That would guide a model better. I would in fact introduce a specific exception for those cases.

Besides other tools raise exception in case of error.

---

Yes agree with this!